services/filesstore/s3store_test.go

```diff
@@ -7,26 +7,24 @@ import (
 	"testing"
 
 	"github.com/mattermost/mattermost-server/model"
+	"github.com/stretchr/testify/require"
 )
 
 func TestCheckMandatoryS3Fields(t *testing.T) {
 	cfg := model.FileSettings{}
 
 	err := CheckMandatoryS3Fields(&cfg)
-	if err == nil || err.Message != "api.admin.test_s3.missing_s3_bucket" {
-		t.Fatal("should've failed with missing s3 bucket")
-	}
+	res := err == nil || err.Message != "api.admin.test_s3.missing_s3_bucket"
+	require.False(t, res, "should've failed with missing s3 bucket")
 
 	cfg.AmazonS3Bucket = model.NewString("test-mm")
 	err = CheckMandatoryS3Fields(&cfg)
-	if err != nil {
-		t.Fatal("should've not failed")
-	}
+	require.Nil(t, err, "should've not failed")
 
 	cfg.AmazonS3Endpoint = model.NewString("")
 	err = CheckMandatoryS3Fields(&cfg)
-	if err != nil || *cfg.AmazonS3Endpoint != "s3.amazonaws.com" {
-		t.Fatal("should've not failed because it should set the endpoint to the default")
-	}
+
+	res = err != nil || *cfg.AmazonS3Endpoint != "s3.amazonaws.com"
```

Comment:
Same as before, you can use `require.Nil(t, err)` and a `require.Equal(t, *cfg.AmazonS3Endpoint, "s3.amazonaws.com")`

Reply:
Applied 👍